File: snippets/react-native/fiat_currencies.ts
import { BindingLiquidSdk } from "@breeztech/breez-sdk-liquid-react-native"

const exampleListCurrencies = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: list-fiat-currencies
  const fiatCurrencies = sdk.listFiatCurrencies()
  // ANCHOR_END: list-fiat-currencies
}

const exampleFetchRates = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: fetch-fiat-rates
  const fiatRates = sdk.fetchFiatRates()
  // ANCHOR_END: fetch-fiat-rates
}

File: snippets/swift/BreezSDKExamples/Sources/FiatCurrencies.swift
import BreezLiquidSDK
import Foundation

func listSupportedFiatCurrencies(sdk: BindingLiquidSdk) -> [FiatCurrency]? {
    // ANCHOR: list-fiat-currencies
    let supportedFiatCurrencies = try? sdk.listFiatCurrencies()
    // ANCHOR_END: list-fiat-currencies
    return supportedFiatCurrencies
}

func getCurrentRates(sdk: BindingLiquidSdk) -> [Rate]? {
    // ANCHOR: fetch-fiat-rates
    let fiatRates = try? sdk.fetchFiatRates()
    // ANCHOR_END: fetch-fiat-rates
    return fiatRates
}
